runners/direct-java/src/main/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactory.java

```diff
@@ -83,7 +83,8 @@ public void cleanup() {}
       if (sourceEvaluators.putIfAbsent(transform, evaluatorQueue) == null) {
         // If no queue existed in the evaluators, add an evaluator to initialize the evaluator
         // factory for this transform
-        BoundedSource<OutputT> source = transform.getTransform().getSource();
+        Bounded<OutputT> bound = (Bounded<OutputT>) transform.getTransform();
```

Comment:
Yeah, changing the `AppliedPTransform` type signature seems to be a bigger rabbit-hole than I expected. Let's leave this one for now. I'm still looking into some of the other changes for this PR.